Add Prometheus metrics for pipeline step duration and failures (#6738)
woodpecker-ci/woodpecker: docs/docs/30-administration/10-configuration/10-server.md

# HELP woodpecker_worker_count Total number of workers.
# TYPE woodpecker_worker_count gauge
woodpecker_worker_count 4
# HELP woodpecker_step_failures_total Total number of pipeline step failures.
# TYPE woodpecker_step_failures_total counter
woodpecker_step_failures_total{pipeline="1",repo="woodpecker-ci/woodpecker",step="deploy"} 1
# HELP woodpecker_step_duration_seconds Step duration in seconds.
# TYPE woodpecker_step_duration_seconds gauge
woodpecker_step_duration_seconds{pipeline="1",repo="woodpecker-ci/woodpecker",step="deploy"} 12
```

> Step-level metrics are only exported when `WOODPECKER_STEP_LEVEL_METRICS` is enabled.

Reviewer: ```suggestion
Step-level metrics are exported as long as `WOODPECKER_STEP_LEVEL_METRICS` is not disabled.
```

---

### STEP_LEVEL_METRICS

- Name: `WOODPECKER_STEP_LEVEL_METRICS`

> - Default: `false`

Reviewer: ```suggestion
- Default: `true`
```
